pkg/controller/operators/olm/operator.go

```diff
@@ -793,8 +793,12 @@ func (a *Operator) transitionCSVState(in v1alpha1.ClusterServiceVersion) (out *v
 		}
 
 		// Check if any generated resources are missing
-		if resErr := a.checkAPIServiceResources(out, certs.PEMSHA256); resErr != nil {
-			out.SetPhase(v1alpha1.CSVPhaseFailed, v1alpha1.CSVReasonAPIServiceResourceIssue, resErr.Error(), now)
+		if resErr := a.checkAPIServiceResources(out, certs.PEMSHA256); len(resErr) > 0 {
+			str := []string{}
+			for _, err := range resErr {
```

Comment:
optional: this would be a great place to use the Aggregated Error type from apimachinery:

https://github.com/kubernetes/kubernetes/blob/4992402fa5bee9f7ac6fe931d101d362a4295cea/staging/src/k8s.io/apimachinery/pkg/util/errors/errors.go#L41

You wouldn't have to manually build this string here